scripts/ci/test_plan.py

```diff
@@ -86,7 +86,7 @@ def __repr__(self):
         return "<Tag {}>".format(self.name)
 
 class Filters:
-    def __init__(self, modified_files, pull_request=False, platforms=[]):
+    def __init__(self, modified_files, pull_request=False, platforms=[], no_path_name = False):
```

Comment:
this option was not added into twister yet, so it is a bit premature to add it here.

Reply:
This is why I got all the relevant commits in a single PR before.
I am adding this generalization to allow running test_plan.py in repos other than zephyr. This no_path_name is crucial, without it other changes are pointless, since names from test_plan script won't match the ones twister is creating in regular call.
I don't mind marking this as DNM until it is resolved what to do with https://github.com/zephyrproject-rtos/zephyr/pull/62332

Reply:
the original PR was too big to review and merge all at once. This option to handle path names can always be added later once the relevant PR went in.

Reply:
I refactored the PR including the new name for the arg. Since the --no-detailed-test-id is a part of the main, I believe this conversation is resolved.